react/features/filmstrip/components/native/TileView.js

```diff
@@ -0,0 +1,324 @@
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { onClick } = this.props;
+        const { height, width } = this.state;
+        const rowElements = this._groupIntoRows(
+            this._renderThumbnails(), this._getColumnCount());
+
+        return (
+            <DimensionsDetector
+                onDimensionsChanged = { this._onDimensionsChanged }>
+                <ScrollView
+                    style = {{
+                        ...styles.tileView,
+                        height,
+                        width
+                    }}>
+                    <TouchableWithoutFeedback onPress = { onClick }>
+                        <View
+                            style = {{
+                                ...styles.tileViewRows,
+                                minHeight: height,
+                                minWidth: width
+                            }}>
+                            { rowElements }
+                        </View>
+                    </TouchableWithoutFeedback>
+                </ScrollView>
+            </DimensionsDetector>
+        );
+    }
+
+    /**
+     * Returns how many columns should be displayed for tile view.
+     *
+     * @returns {number}
+     * @private
+     */
+    _getColumnCount() {
+        const participantCount = this.props._participants.length;
+
+        // For narrow view, tiles should stack on top of each other for a lonely
+        // call and a 1:1 call. Otherwise tiles should be grouped into rows of
+        // two.
+        if (isNarrowAspectRatio(this)) {
+            return participantCount >= 3 ? 2 : 1;
+        }
+
+        if (participantCount === 4) {
+            // In wide view, a four person call should display as a 2x2 grid.
+            return 2;
+        }
+
+        return Math.min(3, participantCount);
+    }
+
+    /**
+     * Returns all participants with the local participant at the end.
+     *
+     * @private
+     * @returns {Participant[]}
+     */
+    _getSortedParticipants() {
+        const participants = [];
+        let localParticipant;
+
+        this.props._participants.forEach(participant => {
+            if (participant.local) {
+                localParticipant = participant;
+            } else {
+                participants.push(participant);
+            }
+        });
+
+        if (localParticipant) {
+            participants.push(localParticipant);
+        }
+
+        return participants;
+    }
+
+    /**
+     * Calculate the height and width for the tiles.
+     *
+     * @private
+     * @returns {Object}
+     */
+    _getTileDimensions() {
+        const { _participants } = this.props;
+        const { height, width } = this.state;
+        const columns = this._getColumnCount();
+        const participantCount = _participants.length;
+        const heightToUse = height - 20;
```

Comment:
where does this 20 come from? maybe make it a constant?

Reply:
This is taking away 20 (10 from each side) of the tile view container so that the tiles aren't up against the edge. I've moved the 10 to a constant.